File: 12_login_attempt_tracker/python/login_tracker.py
USERNAME = "admin"
PASSWORD = "1234"

attempts = 0
MAX_ATTEMPTS = 3

while attempts < MAX_ATTEMPTS:
    user = input("Username: ")
    pwd = input("Password: ")

    if user == USERNAME and pwd == PASSWORD:
        print("Login successful.")
        break
    else:
        attempts += 1
        print("Invalid credentials.")

if attempts == MAX_ATTEMPTS:
    print("Account locked due to too many failed attempts.")

File: 12_login_attempt_tracker/cpp/login_tracker.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    const string USERNAME = "admin";
    const string PASSWORD = "1234";
    const int MAX_ATTEMPTS = 3;

    int attempts = 0;
    string user, pass;

    while (attempts < MAX_ATTEMPTS) {
        cout << "Username: ";
        cin >> user;

        cout << "Password: ";
        cin >> pass;

        if (user == USERNAME && pass == PASSWORD) {
            cout << "Login successful.\n";
            break;
        } else {
            attempts++;
            cout << "Invalid credentials.\n";
        }
    }

    if (attempts == MAX_ATTEMPTS) {
        cout << "Account locked due to too many failed attempts.\n";
    }

    return 0;
}
